### PR TITLE
[SPARK-40993][SPARK-41705][CONNECT] Move Spark Connect documentation and script to dev/ and Python documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR takes over https://github.com/apache/spark/pull/39211 and https://github.com/apache/spark/pull/38477 that proposes:

- Move `connector/connect/dev/generate_protos.sh` → `dev/generate_protos.sh` to be consistent with other places
- Move Python-specific development guides into `python/docs/source/development/testing.rst`

### Why are the changes needed?

To keep the project structure and documentation consistent.

### Does this PR introduce _any_ user-facing change?

Python-specific development guides for Spark Connect will be added in https://spark.apache.org/docs/latest/api/python/development/testing.html.

### How was this patch tested?

I manually tested:

```
./dev/generate_protos.sh
./dev/check-codegen-python.py
```

I also manually verified the Python documentation.

### CODE REVIEW

File: connector/connect/README.md [hunk 1]
```diff
@@ -48,56 +47,3 @@ export CONNECT_PLUGIN_EXEC_PATH=/path-to-protoc-gen-grpc-java-exe
...
-# Run all Spark Connect Python tests as a module.
-./python/run-tests --module pyspark-connect --parallelism 1
-```
-
-
-## Development Topics
-
-### Generate proto generated files for the Python client
-1. Install `buf version 1.11.0`: https://docs.buf.build/installation
-2. Run `pip install grpcio==1.48.1 protobuf==3.19.5 mypy-protobuf==3.3.0 googleapis-common-protos==1.56.4 grpcio-status==1.48.1`
```
Comment (HyukjinKwon) [inline connector/connect/README.md:93]: This is already documented in https://spark.apache.org/docs/latest/api/python/development/contributing.html#environment-setup

File: connector/connect/README.md [hunk 2]
```diff
@@ -48,56 +47,3 @@ export CONNECT_PLUGIN_EXEC_PATH=/path-to-protoc-gen-grpc-java-exe
...
-```bash
-# Run all Spark Connect Python tests as a module.
-./python/run-tests --module pyspark-connect --parallelism 1
-```
-
-
-## Development Topics
-
-### Generate proto generated files for the Python client
-1. Install `buf version 1.11.0`: https://docs.buf.build/installation
```
Comment (HyukjinKwon) [inline connector/connect/README.md:92]: Moved to Environment Setup

File: connector/connect/README.md [hunk 3]
```diff
@@ -48,56 +47,3 @@ export CONNECT_PLUGIN_EXEC_PATH=/path-to-protoc-gen-grpc-java-exe
 The user-defined `protoc` and `protoc-gen-grpc-java` binary files can be produced in the user's compilation environment by source code compilation, 
 for compilation steps, please refer to [protobuf](https://github.com/protocolbuffers/protobuf) and [grpc-java](https://github.com/grpc/grpc-java).
 
-
-### Run Spark Shell
```
Comment (HyukjinKwon) [inline connector/connect/README.md:52]: Move to `python/docs/source/development/testing.rst`

**Comment by HyukjinKwon** [comment]
cc @grundprinzip @amaliujia @zhengruifeng @tedyu FYI

**Review by zhengruifeng — APPROVED** [review]

**Comment by grundprinzip** [comment]
Maybe at least prefix the scripts with connect so that it's clearer where they belong to and not just a random code dump?

**Comment by HyukjinKwon** [comment]
SGTM. let me rename them

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**